projects/plugins/jetpack/modules/sharedaddy/services/class-jetpack-mastodon-modal.php

```diff
@@ -0,0 +1,195 @@
+			&& isset( $_GET['nb'] )
+			&& 1 === (int) $_GET['nb']
+			// phpcs:enable
+		) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Hook the modal render into WordPress.
+	 */
+	public static function modal() {
+		if ( ! self::should_display_modal() ) {
+			return;
+		}
+
+		// Render the modal.
+		self::render_modal();
+
+		die();
+	}
+
+	/**
+	 * Render the modal.
+	 */
+	public static function render_modal() { ?>
+<!doctype html>
+<html <?php language_attributes(); ?>>
+<head>
+	<meta charset="<?php bloginfo( 'charset' ); ?>" />
+	<meta name="viewport" content="width=device-width, initial-scale=1" />
+	<meta name="robots" content="noindex,follow" />
+	<meta name="referrer" content="no-referrer">
+		<?php rel_canonical(); ?>
+	<title><?php esc_html_e( 'Share to Mastodon', 'jetpack' ); ?></title>
+	<style type="text/css">
+		.sd-mastodon-modal {
+			background-color: #282c37;
+			color: #ffffff;
+			display: flex;
+			flex-direction: column;
+			font-family: sans-serif;
+			margin-top: 1em;
+		}
+
+		.sd-mastodon-modal__inner {
+			margin: 0 auto;
+			width: 450px;
+		}
+
+		.sd-mastodon-modal__title,
+		.sd-mastodon-modal__logo {
+			display: flex;
+			font-size: 0;
+			justify-content: center;
+			align-items: center;
+		}
+
+		.sd-mastodon-modal__logo {
+			margin-top: 50px;
+		}
+
+		.sd-mastodon-modal__logo svg {
+			height: 42px;
+		}
+
+		.sd-mastodon-modal__inner form {
+			margin: 2em 0;
+			max-width: 400px;
+			padding: 0 1em;
+		}
+
+		.sd-mastodon-modal__inner label {
+			font-size: 14px;
+			font-weight: 500;
+		}
+
+		.sd-mastodon-modal__inner input {
+			box-sizing: border-box;
+			font-size: 16px;
+			color: #fff;
+			display: block;
+			width: 100%;
+			outline: 0;
+			font-family: inherit;
+			resize: vertical;
+			background: #131419;
+			border: 1px solid #0a0b0e;
+			border-radius: 4px;
+			margin: 15px 10px 10px 0;
+			padding: 10px;
+		}
+
+		.sd-mastodon-modal__inner input:focus {
+			border-color: #8c8dff;
+			background: #17191f;
+		}
+
+		.sd-mastodon-modal__submit button {
+			display: block;
+			width: 100%;
+			border: 0;
+			border-radius: 4px;
+			background: #595aff;
+			color: #fff;
+			font-size: 18px;
+			line-height: inherit;
+			height: auto;
+			margin: 15px 10px 10px 0;
+			padding: 10px;
+			text-decoration: none;
+			text-transform: uppercase;
+			text-align: center;
+			box-sizing: border-box;
+			cursor: pointer;
+			font-weight: 500;
+			outline: 0;
+		}
+
+		.sd-mastodon-modal__submit button:hover {
+			background-color: #6364ff;
+		}
+	</style>
+</head>
+<body class="sd-mastodon-modal">
+	<div class="sd-mastodon-modal__inner">
+		<h2 class="sd-mastodon-modal__title"><?php esc_html_e( 'Share to Mastodon', 'jetpack' ); ?></h2>
+		<div class="sd-mastodon-modal__logo">
+			<?php echo self::$svg; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- SVG is safe. ?>
+		</div>
+		<form action="" method="post">
```

Comment:
The `action` attribute [shouldn't be empty in HTML5](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fs-action). If you want it to submit back to the same page, it should just be omitted.

Back in HTML 4 it was required and an empty string was ok, but the doctype used here is an HTML5 doctype.

Reply:
Done in c1197e23a8ab9573874676c65a4e3718dc30c91e